plugin/storage/factory.go

```diff
@@ -338,37 +338,34 @@ func (f *Factory) initDownsamplingFromViper(v *viper.Viper) {
 	f.FactoryConfig.DownsamplingHashSalt = v.GetString(downsamplingHashSalt)
 }
 
-func (f *Factory) createArchiveSpanReader() (spanstore.Reader, error) {
-	factory, ok := f.archiveFactories[f.SpanReaderType]
-	if !ok {
-		return nil, fmt.Errorf("no %s backend registered for span store", f.SpanReaderType)
-	}
-	return factory.CreateSpanReader()
+type ArchiveStorage struct {
+	Reader spanstore.Reader
+	Writer spanstore.Writer
 }
 
-func (f *Factory) createArchiveSpanWriter() (spanstore.Writer, error) {
-	factory, ok := f.archiveFactories[f.SpanWriterTypes[0]]
+func (f *Factory) InitArchiveStorage() (*ArchiveStorage, error) {
```

Comment:
Why not just return 3 values?

Reply:
@yurishkuro I think its generally a pattern that is avoided in Go. I'm happy to change it if you like that better.